haystack/document_stores/pinecone.py

```diff
@@ -0,0 +1,682 @@
+        a large number of documents without having to load all documents in memory.
+
+        :param index: Name of the index to get the documents from. If None, the
+                      DocumentStore's default index (self.index) will be used.
+        :param filters: Optional filters to narrow down the documents to return.
+                        Example: {"genre": {"$in": ["documentary", "action"]}},
+                        more info on filtering syntax here https://www.pinecone.io/docs/metadata-filtering/
+        :param return_embedding: Whether to return the document embeddings.
+        :param batch_size: When working with large number of documents, batching can help reduce memory footprint.
+        """
+        if headers:
+            raise NotImplementedError("PineconeDocumentStore does not support headers.")
+        if filters:
+            raise Exception("get_all_documents_generator does not support filters.")
+        self._limit_check(batch_size)
+
+        index = index or self.index
+        index = self._sanitize_index_name(index)
+        documents = super(PineconeDocumentStore, self).get_all_documents_generator(
+            index=index, filters=filters, batch_size=batch_size, return_embedding=False
+        )
+        if return_embedding is None:
+            return_embedding = self.return_embedding
+
+        for doc in documents:
+            if return_embedding:
+                if doc.meta and doc.meta.get("vector_id") is not None:
+                    res = self.pinecone_indexes[index].fetch(ids=[doc.id])
+                    if res["vectors"].get(doc.id):
+                        doc.embedding = self._convert_pinecone_result_to_document(
+                            result=res["vectors"][doc.id], return_embedding=return_embedding
+                        ).embedding
+            yield doc
+
+    def get_documents_by_id(
+        self,
+        ids: List[str],
+        index: Optional[str] = None,
+        batch_size: int = 32,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> List[Document]:
+        if headers:
+            raise NotImplementedError("PineconeDocumentStore does not support headers.")
+        self._limit_check(batch_size)
+
+        index = index or self.index
+        index = self._sanitize_index_name(index)
+        # get or create index
+        if not self.pinecone_indexes.get(index):
+            self.pinecone_indexes[index] = self._create_index_if_not_exist(
+                vector_dim=self.vector_dim,
+                index=index,
+                metric_type=self.metric_type,
+                replicas=self.replicas,
+                shards=self.shards,
+            )
+        # check there are vectors
+        count = self.get_embedding_count(index)
+        if count == 0:
+            raise Exception("No documents exist, try creating documents with write_embeddings first.")
```

Comment:
I think this should be changed to either write_documents or update_embeddings.

Reply:
updated comment, now reads `"either write_documents or update_embeddings"`